nghttp/idf_component.yml

```diff
@@ -1,4 +1,4 @@
-version: "1.41.0"
+version: "1.50.0"
```

Comment:
Did you check http2_request example from IDF with this version?

Reply:
http2_request example worked fine with this version of nghttp2. However we have to add -Wno-format compile option in CMakeList of nghttp as done in this PR otherwise we get build error.

Also this version have an heavy impact on binary size. Binary size increases by about 11KB